### DISCUSSION TITLE
OpenFAST encountered an error at simulation time 0.3375 of 630 seconds.  Simulation error level: FATAL ERROR

### DISCUSSION THREAD

**Discussion by DonaghMcN** [OpenFAST/openfast#1467] [Q&A]
Dear OpenFAST software forum,

I am a Master's student in structural engineering and I am currently working on a project that involves the use of OpenFAST software for wind turbine simulation. However, I have been encountering some errors while running my simulations and I am hoping to get some help in resolving this issue.

I have attached the code below for reference:

_The error message I am receiving is (last line):

**OpenFAST input file heading:
    FAST Certification Test #19: NREL 5.0 MW Baseline Wind Turbine with OC3 Monopile RF
    Configuration, for use in offshore analysis

Running ElastoDyn.
Nodal outputs section of ElastoDyn input file not found or improperly formatted.
Running AeroDyn.
AD15 Nodal Outputs: Nodal output section of AeroDyn input file not found or improperly formatted.
Skipping nodal outputs.
Warning: Turning off Unsteady Aerodynamics because polar has constant data. (node 1, blade 1)
Warning: Turning off Unsteady Aerodynamics because polar has constant data. (node 2, blade 1)
Warning: Turning off Unsteady Aerodynamics because polar has constant data. (node 3, blade 1)
Warning: Turning off Unsteady Aerodynamics because polar has constant data. (node 4, blade 1)
Warning: Turning off Unsteady Aerodynamics because polar has constant data. (node 1, blade 2)
Warning: Turning off Unsteady Aerodynamics because polar has constant data. (node 2, blade 2)
Warning: Turning off Unsteady Aerodynamics because polar has constant data. (node 3, blade 2)
Warning: Turning off Unsteady Aerodynamics because polar has constant data. (node 4, blade 2)
Warning: Turning off Unsteady Aerodynamics because polar has constant data. (node 1, blade 3)
Warning: Turning off Unsteady Aerodynamics because polar has constant data. (node 2, blade 3)
Warning: Turning off Unsteady Aerodynamics because polar has constant data. (node 3, blade 3)
Warning: Turning off Unsteady Aerodynamics because polar has constant data. (node 4, blade 3)
Running InflowWind.

   Reading a 31x31 grid (145 m wide, 17.5 m to 162.5 m above ground) with a characteristic wind
   speed of 12 m/s. This full-field file was generated by TurbSim (from OpenFAST-v3.4.0) on
   23-Feb-2023 at 15:02:28.

   Processed 51367 time steps of 80-Hz full-field data (642.08 seconds).
Running HydroDyn.
 Generating incident wave kinematics and current time history.
 Calculating second order difference frequency wave kinematics.
 Calculating second order sum frequency wave kinematics.
Running SubDyn.
   Fixed bottom case detected
   Performing Craig-Bampton reduction 60 DOFs -> 0 modes + 6 DOFs
   Using static improvement method for gravity and ext. loads
   Calculating Full System Modes for output files
   Exporting Summary file
Running ServoDyn.
Running StrucCtrl.
Running ServoDyn Interface for Bladed Controllers (using Intel Visual Fortran for Windows).
Using legacy Bladed DLL interface.
 Time: 0 of 630 seconds.
Warning: SkewedWakeCorrection encountered a large value of chi (98.344 deg), so the yaw
correction will be limited. This warning will not be repeated though the condition may persist.
See the AD15 chi output channels, and consider turning off the Pitt/Peters skew model (set
SkewMod=1) if this condition persists.

FAST_Solution:FAST_AdvanceStates:ED_ABM4:ED_CalcContStateDeriv:SetCoordSy:Small angle assumption
violated in SUBROUTINE SmllRotTrans() due to a large tower deflection (ElastoDyn SetCoordSy). The
solution may be inaccurate. Simulation continuing, but future warnings from SmllRotTrans() will
be suppressed.
 Additional debugging message from SUBROUTINE SmllRotTrans(): 0.3125 s

FAST_Solution:FAST_AdvanceStates:AD_UpdateStates:BEMT_UpdateStates:UpdatePhi(node 5, blade
1):BEMT_UnCoupledSolve:There is no valid value of phi for these operating conditions: Vx =
-0.22308, Vy = -11.98, rlocal = 11.745, theta = 0.23226, geometric phi = -3.123. This warning
will not be repeated though the condition may persist. (See GeomPhi output channel.)

The BEM solution is being turned off due to low TSR.  (TSR = -0.80886). This warning will not be
repeated though the condition may persist. (See GeomPhi output channel.)

FAST_Solution:FAST_AdvanceStates:AD_UpdateStates:BEMT_UpdateStates(node 5, blade
1):UA_UpdateStates:UA_UpdateDiscOtherState:ComputeKelvinChain:Mach number exceeds 1.0. Equations
cannot be evaluated.

 OpenFAST encountered an error at simulation time 0.3375 of 630 seconds.
 Simulation error level: FATAL ERROR

 Aborting OpenFAST.**_

I have tried modifying my input file and adjusting some of the parameters, but I still seem to be getting the same error. Can anyone please help me figure out what I am doing wrong and how I can fix this issue?

Thank you in advance for your help.

Best regards,

Donagh McNamara

**Comment by jjonkman** [comment]
Dear @DonaghMcN,

It sounds like your OpenFAST is suffering a numerical or physical instability.  It looks like you are simulating the OpenFAST r-test of the NREL 5-MW baseline turbine atop the OC3-monopile: https://github.com/OpenFAST/r-test/tree/main/glue-codes/openfast/5MW_OC3Mnpl_DLL_WTurb_WavesIrr.  I would expect this model to run without changes, but my guess is that you changed something in the model.  Can you clarify what you changed?

FYI: Similar instabilities have been discussed many times on our forum: https://forums.nrel.gov/.  I suggest you review similar Q&A there.

Best regards,

**Reply by DonaghMcN** [reply]
Hello,

I have already looked through similar questions and answers on this forum, but I'm still having trouble fixing the errors. The only modifications I've made are to the Wave Height (1.4) and period (6.5), and I've adjusted the time step in the .fst file to match the time step used in the IEC_Crushing files. Additionally, I have included a damper in the ServoDyn file, as shown below (I have also copy and pasted the Baseline STC file into the 5MW_OC3Mnpl_DLL_WTurb_WavesIrr folder. Despite my efforts, I can't seem to identify anything that could be causing the numerical or physical instability I'm experiencing. I was wondering if anyone has any ideas as to what might be causing this issue, as I have been unable to find a solution on this forum. Thank you in advance for your help.

Best regards,

Donagh

SERVODYN FILE
0             NumBStC      - Number of blade structural controllers (integer)
"unused"      BStCfiles    - Name of the files for blade structural controllers (quoted strings) [unused when NumBStC==0]
0             NumNStC      - Number of nacelle structural controllers (integer)
"unused"      NStCfiles    - Name of the files for nacelle structural controllers (quoted strings) [unused when NumNStC==0]
1             NumTStC      - Number of tower structural controllers (integer)
"NRELOffshrBsline5MW_ServoDyn_StC.dat"      TStCfiles    - Name of the files for tower structural controllers (quoted strings) [unused when NumTStC==0]
0             NumSStC      - Number of substructure structural controllers (integer)
"unused"      SStCfiles    - Name of the files for substructure structural controllers (quoted strings) [unused when NumSStC==0]
---------------------- CABLE CONTROL -------------------------------------------

**Reply by jjonkman** [reply]
Dear @DonaghMcN,

Did you increase the time step (`DT`) in the OpenFAST glue code to 0.1 s, which is the time step of the IEC_Crushing file: https://github.com/OpenFAST/r-test/blob/main/glue-codes/openfast/5MW_Baseline/NRELOffshrBsline5MW_Monopile_IEC_Crushing.inp?  If so, this is likely the cause of your instability.  A 0.1 s time step is too large for this NREL 5-MW baseline model unless you've disable degrees of freedom and other functionality.

Best regards,
